### PR TITLE
Add sample for managed application with custom provider 

### PR DESCRIPTION
**Pull request by raosuhas**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [ ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

**Comment by bmoore-msft** [comment]
@raosuhas - can you check the CI failure?

File: 201-managed-application-with-customprovider/README.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+# Extending a Managed Application with a Custom Provider
```
Comment (bmoore-msft) [inline 201-managed-application-with-customprovider/README.md]: add badges

File: 201-managed-application-with-customprovider/artifacts/ManagedAppZip/createUiDefinition.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        "elements": [
+          {
+            "name": "funcname",
+            "type": "Microsoft.Common.TextBox",
+            "label": "Name of the function to be created",
+            "toolTip": "Name of the function to be created",
+            "visible": true
+          },
+          {
+            "name": "storagename",
```
Comment (bmoore-msft) [inline 201-managed-application-with-customprovider/artifacts/ManagedAppZip/createUiDefinition.json]: use the storageAccount control - or just generate a unique name in the template

File: 201-managed-application-with-customprovider/artifacts/ManagedAppZip/mainTemplate.json [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+            "metadata": {
+                "description": "Location for the resources."
+            }
+        },
+        "funcname": {
+            "type": "string",
+            "metadata": {
+                "description": "Name of the azure fuction that hosts the code. Must be globally unique"
+              },
+            "defaultValue": ""
```
Comment (bmoore-msft) [inline 201-managed-application-with-customprovider/artifacts/ManagedAppZip/mainTemplate.json]: remove this or use a globally unique default

File: 201-managed-application-with-customprovider/artifacts/ManagedAppZip/mainTemplate.json [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+        },
+        "storageName": {
+            "type": "string",
+            "metadata": {
+                "description": "Name of the storage account that hosts the function. Must be globally unique"
+              },
+            "defaultValue": ""
```
Comment (bmoore-msft) [inline 201-managed-application-with-customprovider/artifacts/ManagedAppZip/mainTemplate.json]: remove this or use a globally unique default

File: 201-managed-application-with-customprovider/artifacts/ManagedAppZip/mainTemplate.json [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+        },
+        "zipFileBlobUri": {
+          "type": "string",
+          "metadata": {
+            "description": "The Uri for the uploaded function zip file"
+          },
+          "defaultValue": "https://github.com/Azure/azure-quickstart-templates/blob/master/201-managed-application-with-customprovider/artifacts/functionzip/functionpackage.zip?raw=true"
```
Comment (bmoore-msft) [inline 201-managed-application-with-customprovider/artifacts/ManagedAppZip/mainTemplate.json]: See [BP guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#deployment-artifacts-nested-templates-scripts) for how to handle artifacts

File: 201-managed-application-with-customprovider/artifacts/ManagedAppZip/mainTemplate.json [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+    },
+    "variables": {
+        "customrpApiversion" : "2018-09-01-preview",
+        "customProviderName" : "public",
+        "serverFarmName" : "functionPlan"
+    },
+    "resources": [
+        {
+            "type": "Microsoft.Web/serverfarms",
+            "apiVersion": "2016-09-01",
```
Comment (bmoore-msft) [inline 201-managed-application-with-customprovider/artifacts/ManagedAppZip/mainTemplate.json]: update apiVersions throughout

File: 201-managed-application-with-customprovider/artifacts/ManagedAppZip/createUiDefinition.json [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+            "type": "Microsoft.Common.TextBox",
+            "label": "Name of the storage to be created",
+            "toolTip": "Name of the storage to be created",
+            "visible": true
+          }
+        ]
+      }
+    ],
+    "outputs": {
```
Comment (bmoore-msft) [inline 201-managed-application-with-customprovider/artifacts/ManagedAppZip/createUiDefinition.json]: add location to the outputs, use the resourceTypes property to filter the locations you need

File: 201-managed-application-with-customprovider/artifacts/ManagedAppZip/mainTemplate.json [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+            "identity": {
+                "type": "SystemAssigned"
+            },
+            "properties": {
+                "name": "[parameters('funcname')]",
+                "siteConfig": {
+                    "appSettings": [
+                        {
+                            "name": "AzureWebJobsDashboard",
+                            "value": "[concat('DefaultEndpointsProtocol=https;AccountName=',parameters('storageName'),';AccountKey=',listKeys(resourceId('Microsoft.Storage/storageAccounts', parameters('storageName')), '2015-05-01-preview').key1)]"
```
Comment (bmoore-msft) [inline 201-managed-application-with-customprovider/artifacts/ManagedAppZip/mainTemplate.json]: update apiVersions in list*() here and below

File: 201-managed-application-with-customprovider/artifacts/ManagedAppZip/mainTemplate.json [hunk 9]
```diff
@@ -0,0 +1,156 @@
...
+            "sku": {
+                "name": "Standard_LRS"
+            }
+        },
+        {
+            "apiVersion": "[variables('customrpApiversion')]",
+            "type": "Microsoft.CustomProviders/resourceProviders",
+            "name":"[variables('customProviderName')]",
+            "location": "[parameters('location')]",
+            "properties" :{
```
Comment (bmoore-msft) [inline 201-managed-application-with-customprovider/artifacts/ManagedAppZip/mainTemplate.json]: format the JSON (e.g. SHIFT+ALT+F in VS Code)

File: 201-managed-application-with-customprovider/azuredeploy.json [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+            "defaultValue": ""
+        }
+    },
+    "variables": {
+        "locklevel" : "None",
+        "description" : "Sample Managed application definition with custom resources and actions",
+        "displayName" : "Sample Managed application definition with custom resources",
+        "managedappdefname": "[concat(parameters('appname'), '_ApplicationDefinition')]",
+        "packageFileUri":"[uri(parameters('_artifactsLocation'), concat('artifacts/ManagedAppZip/pkg.zip', parameters('_artifactsLocationSasToken')))]",
+        "managedResourceGroupId" : "[concat(subscription().id,'/resourceGroups/', concat(parameters('appname'),'_managed'))]",
```
Comment (bmoore-msft) [inline 201-managed-application-with-customprovider/azuredeploy.json]: remove concat and use subscriptionResourceId()

File: 201-managed-application-with-customprovider/azuredeploy.json [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+                    },
+                    "storageName": {
+                        "value": "[uniqueString(parameters('appname'), resourceGroup().id)]"
+                    },
+                    "zipFileBlobUri":{
+                        "value":"[uri(parameters('_artifactsLocation'), concat('artifacts/functionzip/functionpackage.zip', parameters('_artifactsLocationSasToken')))]"
+                    }
+                }
+            },
+            "dependsOn": [
```
Comment (bmoore-msft) [inline 201-managed-application-with-customprovider/azuredeploy.json]: move this above properties

File: 201-managed-application-with-customprovider/azuredeploy.json (hunk 10 above)
Comment (raosuhas) [reply 201-managed-application-with-customprovider/azuredeploy.json]: That does not work either cause that function needs resource type. I tried doing : 
"managedResourceGroupId": "[subscriptionResourceId('resourceGroups', concat(parameters('applicationName'),'_managed'))]",

But that fails with error : 

 _Unable to evaluate template language function 'subscriptionResourceId': function requires exactly one multi-segmented argument which must be resource type including resource
provider namespace_

**Comment by bmoore-msft** [comment]
Closing abandoned request - please reopen if you can address the feedback.

**Pull request closed without merging**